src/chrome/chromeUtils.ts

```diff
@@ -312,3 +314,46 @@ export function getUrlRegexForBreakOnLoad(url: string): string {
     const escapedFileName = pathToRegex(fileNameWithoutExtension);
     return '.*[\\\\\\/]' + escapedFileName + '([^A-z^0-9].*)?$';
 }
+
+/**
+ * Checks if a given tcp port is currently in use (more accurately, is there a server socket accepting connections on that port)
+ * @param port The port to check
+ * @param host Optional host, defaults to 127.0.0.1
+ * @param timeout Timeout for the socket connect attempt
+ * @returns True if a server socket is listening on the given port, false otherwise
+ */
+export async function isPortInUse(port: number, host = '127.0.0.1', timeout = 400): Promise<boolean> {
+    // Basically just create a socket and try to connect on that port, if we can connect, it's open
+    return new Promise<boolean>((resolve, _reject) => {
+        const socket = new Socket();
+        const callback = (inUse: boolean) => () => {
+            resolve(inUse);
+            socket.removeAllListeners();
+            socket.destroy();
+        };
+
+        socket.setTimeout(timeout);
+        socket.on('connect', callback(true));
```

Comment:
Won't this call callback immediately, not on the event?

Reply:
Sorry, I realize it wasn't super clear that the `callback` function was actually creating a new function., and _that_ function was actually being passed as the callback. Changed to be clearer.

Reply:
Oh I didn't read closely. Thanks.